src/i2c/src/i2c.rs

```diff
@@ -223,6 +239,11 @@ impl I2cSmbusIoctlData {
             data: &mut data,
         })
     }
+
+    fn data_ref(&self) -> &I2cSmbusData {
+        // Safe as this gets the reference to a valid pointer
```

Comment:
I don't think this is actually safe because you cannot know if the object lives as long as the reference. The fix for this safety issue is most likely not as simple as writing one line of code.

I would suggest to not work with raw pointers because that will make the code inherently unsafe. This requires changes in how `I2cSmbusIoctlData` is defined, implemented and later on used.

Reply:
We can do it in another PR, but then we should remove the "fixes #33" so that the issue stays open.

Reply:
Done